source/app/configuration.py

```diff
@@ -53,6 +65,94 @@ def validate_config(self):
                 raise IrisConfigException(
                     'Missing section %s in the configuration file' % section)
 
+    def config_key_vault(self):
+        """
+        Load the settings to connect to Azure Key Vault
+        """
+
+    def load(self, section, option, fallback=None):
+        """
+        Load variable from different sources. Uses the following order
+        1. Azure Key Vault
+        2. Environment Variable
+        3. Environment Variable deprectated
+        3. Configuration File
+        """
+
+        loaders = [self._load_azure_key_vault,
+                   self._load_env, self._load_env_deprecated,
+                   self._load_file, self._load_file_deprecated]
+        for loader in loaders:
+            value = loader(section, option)
+            if value:
+                return value
+
+        return fallback
+
+    def _load_azure_key_vault(self, section, option):
+        if not (hasattr(self, 'key_vault_name') and self.key_vault_name):
+            return
+
+        key = f"{section}{option}".replace('_', '')
+
+        try:
+            return self.az_client.get_secret(key).value
+        except ResourceNotFoundError:
+            return None
+
+    def _load_env(self, section, option):
+        return os.environ.get(f"{section}_{option}")
+
+    def _load_env_deprecated(self, section, option):
+        # Specify new_value : old_value
+        mapping = {
+            'POSTGRES_USER': 'DB_USER',
+            'POSTGRES_PASSWORD': 'DB_PASS',
+            'POSTGRES_ADMIN_USER': 'POSTGRES_USER',
+            'POSTGRES_HOST': 'DB_HOST',
+            'POSTGRES_PORT': 'DB_PORT',
```

Comment:
This mapping should probably be `POSTGRES_SERVER` otherwise the app will not see the already set old `DB_HOST`. 

```suggestion
            'POSTGRES_SERVER': 'DB_PORT',
```